src/common/strtol.cc

```diff
@@ -28,6 +28,17 @@ long long strict_strtoll(const char *str, int base, std::string *err)
   errno = 0; /* To distinguish success/failure after call (see man page) */
   long long ret = strtoll(str, &endptr, base);
 
+  if (endptr == str
```

Comment:
can we just put
```c++
if (endptr == str && ret == 0) {
 ...
```

see http://pubs.opengroup.org/onlinepubs/9699919799/functions/strtol.html,
> If the subject sequence is empty or does not have the expected form, no conversion is performed; the value of nptr shall be stored in the object pointed to by endptr, provided that endptr is not a null pointer.

and 
> Upon successful completion, these functions shall return the converted value, if any. If no conversion could be performed, 0 shall be returned [CX] [Option Start]  and errno **may** be set to [EINVAL]. [Option End]